mono/metadata/mempool.c

```diff
@@ -317,13 +370,104 @@ mono_mempool_alloc (MonoMemPool *pool, guint size)
 
 			rval = pool->pos;
 			pool->pos += size;
+
 		}
 	}
 #endif
 
+#ifdef ALLOCATION_WITH_CANARIES
+	AllocRecord *curr_record = g_malloc (sizeof (AllocRecord));
+	curr_record->object_start = (guint8 **)rval;
+	curr_record->padding_start = (intptr_t)curr_record->object_start + requested_size;
+	guint8 **addr;
+
+	for (addr = curr_record->padding_start; addr < curr_record->padding_start + PADDING_COUNT; addr++)
+	{
+#if 0
+		// fprintf(stderr, "Writing canary to %p\n", addr);
```

Comment:
When it's just a single line like these, the `#if` is kinda useless. My feedback was meant for a block of code below.